### ISSUE TITLE
Error. No naistyles.csv found when connect comfyui web

### ISSUE THREAD

**Issue by eilinge** [comfyanonymous/ComfyUI#3463]
"Error. No naistyles.csv found" when running comfyui, how to fix it? Since importing a production flow about clay and installing the missing nodes, comfyui can start normally, but this error will be reported when connecting to comfyui web.

**Comment by shawnington** [comment]
"naistyles.csv" is not part of comfy it is related to one of your nodes, or is saved in the workflow somehow. Please go through the nodes in your workflow and consult the github's of the appropriate creators.

**Comment by eilinge** [comment]
This problem also occurs when I clear the workflow. This error log does not indicate which node is causing it.

**Comment by eilinge** [comment]
This error indicates that naisytles.csv needs to be placed in the ComfyUI-NAI_mod/CSV node, but the custom node is not found in github.

**Comment by shawnington** [comment]
> This problem also occurs when I clear the workflow. This error log does not indicate which node is causing it.

There is no "naistyles.csv" in the comfy workflow. Unfortunately errors are not always helpful. It is related to one of your nodes, they are created by individual people, and not part of an official repository that adhereds to a rigorous standard. We are looking into improving this however, Ive been told. 

The best we can do for you is tell you to check the github's of all nodes you are using and see if any of them have that file, and if you post which repo and the complete error text, I will try and resolve it, even though it should technically be an issue on the offending nodes GitHub.

Some times people do remove nodes from their packs without depreciating them, and if you updated, you are kinda out of luck unless you manually reinstall the previous version that had the node.

**Comment by mskiller** [comment]
this issue come from ComfyUI-Universal-Styler, uninstall it you should be fine.

**Comment by sugarkwork** [comment]
I am experiencing the same issue.
It seems to be a message from the ComfyUI-Universal-Styler node.
Here is the error output code:

https://github.com/KoreTeknology/ComfyUI-Universal-Styler/blob/main/naistyler_nodes.py#L106

The path mentioned in the error and the actual path it is trying to load are different, and it can be seen that the error message is misleading.

If you want to prevent the error, you can do the following:

1. Create a ComfyUI-NAI-styler directory under the custom_nodes directory.
2. Create an \_\_init\_\_.py file under the ComfyUI-NAI-styler directory with the following content:
```python
__all__ = ['NODE_CLASS_MAPPINGS', 'NODE_DISPLAY_NAME_MAPPINGS']
NODE_CLASS_MAPPINGS = {}
NODE_DISPLAY_NAME_MAPPINGS = {}
```
3. Create a CSV directory under the ComfyUI-NAI-styler directory .
4. Create three empty files under the CSV directory:

    naifilters.csv
    naistyles.csv
    naitypes.csv

After restarting ComfyUI, the error should no longer appear.

I couldn't find out what ComfyUI-NAI-styler is. There seems to be no such node.
This might need to be discussed in the ComfyUI-Universal-Styler repository.

**Comment by jonnytracker** [comment]
still issue going

**Comment by AilinTaban** [comment]
I'm new here so I apologize if I'm wrong on all this... I have also tried to solve these errors...  
I found this link to the Guy who created this CVS issue... found in the 'ComfyUI-Universal-Styler' folder- ReadMe.md file, not sure if this helps! I must have Installed this Node--- 

* Author: **Uriel Deveaud** - [Kore Teknology](https://github.com/KoreTeknology) 
* License: This project is released under the GPL License.
* This work is dedicated to all ComfyUI users around the world ;)https://github.com/KoreTeknology/ComfyUI-Universal-Styler

VERSION: Pre-Alpha Release: 0.2.1 - Licensed under GNU General Public License v3.0
"   !!! This project and its documentation are (always) in progress !!! stay tuned...
- You may find some errors, please report the issue in the dedicated section, thx  "

**Comment by mcmonkey4eva** [comment]
Please report this issue to the developer of the relevant custom nodes

**Comment by jokerfree** [comment]
在你的插件库中把ComfyUI-Universal-Styler文件夹中的所有ComfyUI-NAI-styler替换成ComfyUI-Universal-Styler，就可以解决。猜测是作者上传代码的时候脑子一激灵给他的工程改了个漂亮的名字，却忘记了他代码中写死的文件路径，这是一个不负责任的作者。

**Comment by chinamaler** [comment]
Modify "ComfyUI-NAI-styler" in "naistyler_nodes.py" to "ComfyUI-Universal-Styler"
![屏幕截图 2024-08-04 151950](https://github.com/user-attachments/assets/1ac12d2b-f1a7-4be8-92b9-4f0c08c92def)

![屏幕截图 2024-08-04 152013](https://github.com/user-attachments/assets/20f788bb-6d2d-475c-bfe2-f8640bec4adf)

**Comment by 1MB4R0SS4** [comment]
> chinamaler

thanks, that worked

**Comment by rugabunda** [comment]
> Modify "ComfyUI-NAI-styler" in "naistyler_nodes.py" to "ComfyUI-Universal-Styler" ![屏幕截图 2024-08-04 151950](https://private-user-images.githubusercontent.com/44858367/354883956-1ac12d2b-f1a7-4be8-92b9-4f0c08c92def.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5MTA1OTUsIm5iZiI6MTcyMzkxMDI5NSwicGF0aCI6Ii80NDg1ODM2Ny8zNTQ4ODM5NTYtMWFjMTJkMmItZjFhNy00YmU4LTkyYjktNGYwYzA4YzkyZGVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE3VDE1NTgxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmNjM3OGZmYThkZjBhOTNiOWQ3MzVkMWVhYWU1NzNkMjc0NjkyZjAyOTQyYzcwMDIxMWE4ODI0ZmIxNjg4YzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7uH71tYTZcIUnHUyU43FCwLdTgblu3kSIVQevVHctzU)
> 
> ![屏幕截图 2024-08-04 152013](https://private-user-images.githubusercontent.com/44858367/354884187-20f788bb-6d2d-475c-bfe2-f8640bec4adf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5MTA1OTUsIm5iZiI6MTcyMzkxMDI5NSwicGF0aCI6Ii80NDg1ODM2Ny8zNTQ4ODQxODctMjBmNzg4YmItNmQyZC00NzVjLWJmZTItZjg2NDBiZWM0YWRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE3VDE1NTgxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhNzMxYTNhNGZiMzMyODI4ODcyNWQ0ZGU3NTlmYzM0ZTUwYmY5NjQyN2VmZWU1OTZhY2E3Mzg1YWUzYTU5MTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PY1S8JFinW6wUi5CYDJOCQ0l9p8EfH-2Wd-Xrm3wOZ0)

Did this but I had to repeat this every single time i checked for updates in comfy ui... I disabled this node because its such a mess and the maintainer hasn't fixed it

**Comment by ryanwyler** [comment]
**Here's the real workaround that works for me:** 

1. Open Powershell as Administrator and go to the ComfyUI\custom_nodes folder:
Set-Location "C:\CHANGE_TO_THE_PATH_TO_COMFYUI\ComfyUI\custom_nodes"

2. Create the symlink for "ComfyUI-NAI-styler" pointing to ComfyUI-Universal-Styler
New-Item -ItemType SymbolicLink -Path "ComfyUI-NAI-styler" -Target "ComfyUI-Universal-Styler"

**Why is this happening?  Two problems.** 
1. The module called ComfyUI-Universal-Styler has the wrong path reference to load the csv files: 
LINE NUMBER 148: 
    @classmethod
    def INPUT_TYPES(cls):
        cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-NAI-styler\\CSV\\naifilters.csv"))
        cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-NAI-styler\\CSV\\naistyles.csv"))
        cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-NAI-styler\\CSV\\naitypes.csv"))

2. The error message in load_naistyles_csv is hard coded the wrong path location specifying "ComfyUI_NAI-styler" instead of using the variable naitypes_path in the error message, so that's why it's confusing:
LINE NUMBERS 131 & 140:
            print(f"""Error. No naistyles.csv found. Put your naistyles.csv in the custom_nodes/ComfyUI_NAI-mod/CSV directory of ComfyUI. Then press "Refresh". Your current root directory is: {folder_paths.base_path}""")

**Comment by over12come** [comment]
Do this:

**Comment by FerreiraArmando** [comment]
> **Here's the real workaround that works for me:**
> 
> 1. Open Powershell as Administrator and go to the ComfyUI\custom_nodes folder:
>    Set-Location "C:\CHANGE_TO_THE_PATH_TO_COMFYUI\ComfyUI\custom_nodes"
> 2. Create the symlink for "ComfyUI-NAI-styler" pointing to ComfyUI-Universal-Styler
>    New-Item -ItemType SymbolicLink -Path "ComfyUI-NAI-styler" -Target "ComfyUI-Universal-Styler"
> 
> **Why is this happening? Two problems.**
> 
> 1. The module called ComfyUI-Universal-Styler has the wrong path reference to load the csv files:
>    LINE NUMBER 148:
>    @classmethod
>    def INPUT_TYPES(cls):
>    cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-NAI-styler\CSV\naifilters.csv"))
>    cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-NAI-styler\CSV\naistyles.csv"))
>    cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-NAI-styler\CSV\naitypes.csv"))
> 2. The error message in load_naistyles_csv is hard coded the wrong path location specifying "ComfyUI_NAI-styler" instead of using the variable naitypes_path in the error message, so that's why it's confusing:
>    LINE NUMBERS 131 & 140:
>    print(f"""Error. No naistyles.csv found. Put your naistyles.csv in the custom_nodes/ComfyUI_NAI-mod/CSV directory of ComfyUI. Then press "Refresh". Your current root directory is: {folder_paths.base_path}""")

Works fine! Thanks!

**Comment by adrienchristiaen** [comment]
I don't know why some of the answers here didn't worked for me.
So i just change the file: "naistyler_nodes.py" in ComfyUI/custom_nodes/ComfyUI-Universal-Styler/naistyler_nodes.py

**I change this part:**
"
@classmethod
    def INPUT_TYPES(cls):
        cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naifilters.csv"))
        cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naistyles.csv"))
        cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naitypes.csv"))
        return {
            "required": {
                #"mute": (["On", "Off"],),
                "naifilters": (list(cls.naistyles_csv.keys()),),
                "naistyles": (list(cls.naifilters_csv.keys()),),
                "naitypes": (list(cls.naitypes_csv.keys()),),
                #"clip": ("CLIP", ),
            },                      
        }
"
        
**With this one:**

"
@classmethod
def INPUT_TYPES(cls):
    base_path = Path(folder_paths.base_path)  # Use of Path to ensure path compatibility
    cls.naistyles_csv = cls.load_naistyles_csv(base_path / "custom_nodes/ComfyUI-Universal-Styler/CSV/naistyles.csv")
    cls.naifilters_csv = cls.load_naifilters_csv(base_path / "custom_nodes/ComfyUI-Universal-Styler/CSV/naifilters.csv")
    cls.naitypes_csv = cls.load_naitypes_csv(base_path / "custom_nodes/ComfyUI-Universal-Styler/CSV/naitypes.csv")
    return {
        "required": {
            "naifilters": (list(cls.naistyles_csv.keys()),),
            "naistyles": (list(cls.naifilters_csv.keys()),),
            "naitypes": (list(cls.naitypes_csv.keys()),),
        },                      
    }
"

You can add some print to check if everything's working:

"
print(f"naistyles.csv Path: {base_path / 'custom_nodes/ComfyUI-Universal-Styler/CSV/naistyles.csv'}")
print(f"naifilters.csv Path: {base_path / 'custom_nodes/ComfyUI-Universal-Styler/CSV/naifilters.csv'}")
print(f"naitypes.csv Path: {base_path / 'custom_nodes/ComfyUI-Universal-Styler/CSV/naitypes.csv'}")
"

**Comment by adrienchristiaen** [comment]
I don't know why some of the answers here didn't worked for me.
So i just change the file: "naistyler_nodes.py" in ComfyUI/custom_nodes/ComfyUI-Universal-Styler/naistyler_nodes.py

I change this part:
"
@classmethod
def INPUT_TYPES(cls):
cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-Universal-Styler\CSV\naifilters.csv"))
cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-Universal-Styler\CSV\naistyles.csv"))
cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-Universal-Styler\CSV\naitypes.csv"))
return {
"required": {
#"mute": (["On", "Off"],),
"naifilters": (list(cls.naistyles_csv.keys()),),
"naistyles": (list(cls.naifilters_csv.keys()),),
"naitypes": (list(cls.naitypes_csv.keys()),),
#"clip": ("CLIP", ),
},
}
"

With this one:

"
@classmethod
def INPUT_TYPES(cls):
base_path = Path(folder_paths.base_path) # Use of Path to ensure path compatibility
cls.naistyles_csv = cls.load_naistyles_csv(base_path / "custom_nodes/ComfyUI-Universal-Styler/CSV/naistyles.csv")
cls.naifilters_csv = cls.load_naifilters_csv(base_path / "custom_nodes/ComfyUI-Universal-Styler/CSV/naifilters.csv")
cls.naitypes_csv = cls.load_naitypes_csv(base_path / "custom_nodes/ComfyUI-Universal-Styler/CSV/naitypes.csv")
return {
"required": {
"naifilters": (list(cls.naistyles_csv.keys()),),
"naistyles": (list(cls.naifilters_csv.keys()),),
"naitypes": (list(cls.naitypes_csv.keys()),),
},
}
"

You can add some print in the method to check if everything's working:

"
print(f"naistyles.csv Path: {base_path / 'custom_nodes/ComfyUI-Universal-Styler/CSV/naistyles.csv'}")
print(f"naifilters.csv Path: {base_path / 'custom_nodes/ComfyUI-Universal-Styler/CSV/naifilters.csv'}")
print(f"naitypes.csv Path: {base_path / 'custom_nodes/ComfyUI-Universal-Styler/CSV/naitypes.csv'}")
"

And that worked fine for me !

**Comment by Suka4GitHub** [comment]
This worked for me when nothing else would....strangely the error (for me anyways) appeared when using marascott upscaling node and it impacting rendering times. Thanks a bunch! :-)

**Comment by 933127** [comment]
> 修改“naistyler_nodes.py”中的“ComfyUI-NAI-styler”为“ComfyUI-Universal-Styler” ![屏幕截图 2024-08-04 151950](https://private-user-images.githubusercontent.com/44858367/354883956-1ac12d2b-f1a7-4be8-92b9-4f0c08c92def.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3Njk0OTMsIm5iZiI6MTcyNDc2OTE5MywicGF0aCI6Ii80NDg1ODM2Ny8zNTQ4ODM5NTYtMWFjMTJkMmItZjFhNy00YmU4LTkyYjktNGYwYzA4YzkyZGVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDE0MzMxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiMzk2MDc1ZGRlODAxMDdhNTgwMGJlNTA4YTAzMTJjYTY5NWYyMDVlNTE1NzJmNTViNGVkZGJkZGQyY2EwMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.n1EOA8JbNvpuK2OVP31T1uTvopwh7sOz3CveVhCWXJw)
> 
> ![屏幕截图 2024-08-04 152013](https://private-user-images.githubusercontent.com/44858367/354884187-20f788bb-6d2d-475c-bfe2-f8640bec4adf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3Njk0OTMsIm5iZiI6MTcyNDc2OTE5MywicGF0aCI6Ii80NDg1ODM2Ny8zNTQ4ODQxODctMjBmNzg4YmItNmQyZC00NzVjLWJmZTItZjg2NDBiZWM0YWRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDE0MzMxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjZTg4MDViZjA5NmZlZWYyY2VmYmIxMjU0NDdiZTExMjc0YzRlOWYxZDQwNjliOGUwYWE5NWZiMzZkOTg3ODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.l74iv-MP5LC3X54s0dsFl2HNj6UKBj5-tdAVvxhx8PQ)

This is really a master among the people.

**Comment by mrlunk** [comment]
Solution found on reddit in post:
https://www.reddit.com/r/comfyui/comments/1cqpjkf/comment/ligc0ie/

---------------QUOTE:-----------------------
in \custom_nodes\ComfyUI_NAI-mod

Modify "ComfyUI-NAI-styler" in "naistyler_nodes.py" to "ComfyUI-Universal-Styler"

Search and replace... should be 4 entries

restart comfyui - solved ;-)
--------------------------------------------------

**Comment by occultskyrong** [comment]
just in `custom-nodes` with `CMD`

```
mklink /D ComfyUI-NAI-styler ComfyUI-Universal-Styler

mklink /D ComfyUI_NAI-mod ComfyUI-Universal-Styler
```

**Comment by offmybach** [comment]
> just in `custom-nodes` with `CMD`
> 
> ```
> mklink /D ComfyUI-NAI-styler ComfyUI-Universal-Styler
> 
> mklink /D ComfyUI_NAI-mod ComfyUI-Universal-Styler
> ```

I see your solution, and may be dumb to ask, but what to do with these cmds? what is mklink? obviously have to use correct drive letter but how to apply? TIA

**Comment by beijingrong** [comment]
The above https://github.com/comfyanonymous/ComfyUI/issues/3463#issuecomment-2231823297 and https://github.com/comfyanonymous/ComfyUI/issues/3463#issuecomment-2267394860 worked for me. 

Also, the original code specifies Windows specific paths, which causes the errors on my linux: 
```
cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naifilters.csv"))
cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naistyles.csv"))
cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naitypes.csv"))

```
We can change the line to generic paths: 

```
        cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes", "ComfyUI-Universal-Styler", "CSV", "naifilters.csv"))
        cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes", "ComfyUI-Universal-Styler", "CSV", "naistyles.csv"))
        cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes", "ComfyUI-Universal-Styler", "CSV", "naitypes.csv"))

```

**Comment by Agiltech-com-au** [comment]
> 2\. __init__.py

Thanks, I just searched this file for the wrong path and replaced it with ComfyUI-Universal-Styler. you can even see the error message with the wrong path :P kind of obvious issue. surprised not fixed yet. 

`    @classmethod
    def INPUT_TYPES(cls):
        cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naifilters.csv"))
        cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naistyles.csv"))
        cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\\ComfyUI-Universal-Styler\\CSV\\naitypes.csv"))`

![image](https://github.com/user-attachments/assets/154fd77a-48d7-4970-a8a1-f505cb5db3cd)
![Screenshot 2024-10-16 004700](https://github.com/user-attachments/assets/e7a9823f-3ac8-42bd-81ab-a91de54b5b8b)

**Comment by sugarkwork** [comment]
I investigated why this incomplete ComfyUI-Universal-Styler custom node is needed, and I found one thing:

https://github.com/KoreTeknology/ComfyUI-Universal-Styler/blob/main/naistyler_nodes.py

This code includes the following node definition:

"ShowText|pysssss": "✴️ U-NAI Get Text"

Looking at this code, it's clear that it's a copy of the following code:

https://github.com/pythongosssss/ComfyUI-Custom-Scripts/blob/main/py/show_text.py

It seems that the author of ComfyUI-Universal-Styler copied and pasted the node from ComfyUI-Custom-Scripts and only changed the display name of the node.

If a workflow using ShowText from ComfyUI-Custom-Scripts was distributed, ComfyUI-Universal-Styler might be found as a dependency before the original ComfyUI-Custom-Scripts. As a result, it gets installed as a dependency, leading to encounters with ComfyUI-Universal-Styler errors.

In other words, if you're not explicitly using ComfyUI-Universal-Styler nodes, you could probably uninstall ComfyUI-Universal-Styler and install ComfyUI-Custom-Scripts instead. This should allow the workflow to function normally.

**Comment by Agiltech-com-au** [comment]
> Better make this one (answer for total noobs):
> 
> Windows+R > CMD replace "X:\Comfy_UI" to what your patch to folder:
> 
> `mklink /J "X:\Comfy_UI\Comfy-UI-Manager\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-NAI-styler" "X:\Comfy_UI\Comfy-UI-Manager\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Universal-Styler" `
> 
> p.s. This one big line command, not two. And this will create not shortcut, but a hard linked folder 'ComfyUI-NAI-styler' (that don't exist before) to 'ComfyUI-Universal-Styler'

Actually yeah junction points could be better. Because as soon as I update the styler it's going to possibly put in the wrong directories in the config file again 🤣

**Comment by KINGLIFER** [comment]
> I investigated why this incomplete ComfyUI-Universal-Styler custom node is needed, and I found one thing:
> 
> https://github.com/KoreTeknology/ComfyUI-Universal-Styler/blob/main/naistyler_nodes.py
> 
> This code includes the following node definition:
> 
> "ShowText|pysssss": "✴️ U-NAI Get Text"
> 
> Looking at this code, it's clear that it's a copy of the following code:
> 
> https://github.com/pythongosssss/ComfyUI-Custom-Scripts/blob/main/py/show_text.py
> 
> It seems that the author of ComfyUI-Universal-Styler copied and pasted the node from ComfyUI-Custom-Scripts and only changed the display name of the node.
> 
> If a workflow using ShowText from ComfyUI-Custom-Scripts was distributed, ComfyUI-Universal-Styler might be found as a dependency before the original ComfyUI-Custom-Scripts. As a result, it gets installed as a dependency, leading to encounters with ComfyUI-Universal-Styler errors.
> 
> In other words, if you're not explicitly using ComfyUI-Universal-Styler nodes, you could probably uninstall ComfyUI-Universal-Styler and install ComfyUI-Custom-Scripts instead. This should allow the workflow to function normally.

1000%

**Comment by hieuck** [comment]
https://github.com/KoreTeknology/ComfyUI-Universal-Styler/pull/9/commits/b54b7dc5772be4cf84d072d874654bd6b99b2293

**Comment by Zooloos007** [comment]
> **Here's the real workaround that works for me:**
> 
>     1. Open Powershell as Administrator and go to the ComfyUI\custom_nodes folder:
>        Set-Location "C:\CHANGE_TO_THE_PATH_TO_COMFYUI\ComfyUI\custom_nodes"
> 
>     2. Create the symlink for "ComfyUI-NAI-styler" pointing to ComfyUI-Universal-Styler
>        New-Item -ItemType SymbolicLink -Path "ComfyUI-NAI-styler" -Target "ComfyUI-Universal-Styler"
> 
> 
> **Why is this happening? Two problems.**
> 
>     1. The module called ComfyUI-Universal-Styler has the wrong path reference to load the csv files:
>        LINE NUMBER 148:
>        @classmethod
>        def INPUT_TYPES(cls):
>        cls.naistyles_csv = cls.load_naistyles_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-NAI-styler\CSV\naifilters.csv"))
>        cls.naifilters_csv = cls.load_naifilters_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-NAI-styler\CSV\naistyles.csv"))
>        cls.naitypes_csv = cls.load_naitypes_csv(os.path.join(folder_paths.base_path, "custom_nodes\ComfyUI-NAI-styler\CSV\naitypes.csv"))
> 
>     2. The error message in load_naistyles_csv is hard coded the wrong path location specifying "ComfyUI_NAI-styler" instead of using the variable naitypes_path in the error message, so that's why it's confusing:
>        LINE NUMBERS 131 & 140:
>        print(f"""Error. No naistyles.csv found. Put your naistyles.csv in the custom_nodes/ComfyUI_NAI-mod/CSV directory of ComfyUI. Then press "Refresh". Your current root directory is: {folder_paths.base_path}""")

You cool man! Thank you!

**Comment by Sechini858** [comment]
>>>>>>>>https://github.com/KoreTeknology/ComfyUI-Universal-Styler<<<<<<<It was resolved here, thank you!

**Comment by KoreTeknology** [comment]
Hi everyone, i just see this issue regarding the old version of my custom nodes!
It is all fixed now and it has new features too!

i am sorry to see that you get trouble, but i did mention on my github page that this project was in progress and some pieces of codes were juste pasted in... now, everything is ok and rewrited properly.

Any other problem could be reported, please.
Thank you all, regards
